utils/rtty/files/rtty.init

```diff
@@ -0,0 +1,71 @@
+#!/bin/sh /etc/rc.common
+
+USE_PROCD=1
+START=99
+
+BIN=/usr/sbin/rtty
+
+invalid=
+ifname=
+id=
+description=
+host=
+port=
+
+device_config() {
+    local cfg="$1"
+
+    uci_validate_section rtty device "${1}" \
+        'interface:string:br-lan' \
+        'id:and(string,maxlength(63))' \
+        'description:and(string,maxlength(126))' || invalid=1
+
+    [ -n "$interface" ] && ifname=$(uci -q -p /var/state get network.$interface.ifname)
+}
+
+server_config() {
+    local cfg="$1"
+
+    uci_validate_section rtty "${1}" \
+        'host:host' \
+        'port:port' \
+        'ssl:bool:0' || invalid=1
+}
+
+check_required() {
+    local name=$1
+    local val
+
+    eval val="\$$name"
+    [ -z "$val" ] && {
+        echo "$name required"
+        exit 1
+    }
+}
+
+start_service() {
+    config_load rtty
+    config_foreach device_config device
+    config_foreach server_config server
+
+    if [ $invalid -eq 1 ];
+    then
+        echo "validation failed"
+        exit 1
+    else
```

Comment:
`invalid` can be empty string and it will cause shell to emit error.  `[ -n "$invalid" ]` can be used here.

Error message should be put into stderr

Reply:
Fixed.